### ISSUE TITLE
Assertion failed: Val && "isa<> used on a null pointer"

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-prepo#29]
Compile the following code:
```
template <typename _CharT>
class basic_ostream;

typedef basic_ostream<char> ostream;
template <typename _CharT>
basic_ostream<_CharT> &__ostream_insert(basic_ostream<_CharT> &__out,
                                        _CharT *__s);
inline basic_ostream<char> &operator<<(basic_ostream<char> &__out, char *__s) {
  __ostream_insert(__out, __s);
}

ostream &operator<<(ostream &stream, int version) {	
  char *str = "";
  switch (version) {
  case 1:
    str = "time_based";
    break;
  case 2:
    str;
  case 3:
    str = "name_based_md5";
  }
  return stream << str;
}

```
using
```
$ rm clang.db
$ clang++ -target x86_64-pc-linux-gnu-repo -O2 -std=gnu++11 -c test.cpp
```
Compilation failed with the following error:
```
Assertion failed: Val && "isa<> used on a null pointer", file C:\MyWork\git\github\llvm-prepo\include\llvm/Support/Casting.h, line 106
```

**Comment by MaggieYingYi** [comment]
The SimplifyCFG Pass creates a new global variable named “switch.table.FunctionName“. Since the SimplifyCFG pass is after the RepoTicketGenerator Pass, the new global variable does not contain the TicketNode metadata. It caused the above issue.
 
To fix the issue, a TicketNode metadata is created for the global object that doesn’t contain the TicketNode metadata.

**Comment by MaggieYingYi** [comment]
Fixed in the commit: 3aa3dc42034faf14.
